### PR TITLE
Fixing Blob bindings

### PR DESCRIPTION
**Pull request by mathewc**
These changes fix all our broken Blob samples. There were core issues in the binding pipeline which I've addressed. I've fixed all the samples and tested each - they are now successfully triggered on new blobs, and their out bindings also work (with parameter binding).

There is one ugly hack required (private reflection) until we decide what to do in the core SDK. We might need to make the data we need public in some way.

### CODE REVIEW

**Comment by davidebbo** [comment]
Ah, so that's why we could never get Blob to work? Sweet!

**Comment by mathewc** [comment]
I just pushed this, but I'll address any feedback anyone has. I wanted to get this in.

File: src/WebJobs.Script/Description/ScriptFunctionInvokerBase.cs [hunk 1]
```diff
@@ -64,27 +67,77 @@ protected virtual void OnScriptFileChanged(object sender, FileSystemEventArgs e)
...
+        /// JSON bodies, etc.
+        /// </summary>
+        protected static void ApplyAmbientBindingData(IBinder binder, IDictionary<string, string> bindingData)
+        {
+            // TEMP: Dig the ambient binding data out of the binder
+            FieldInfo fieldInfo = binder.GetType().GetField("_bindingSource", BindingFlags.NonPublic | BindingFlags.Instance);
+            var bindingSource = fieldInfo.GetValue(binder);
+            PropertyInfo propertyInfo = bindingSource.GetType().GetProperty("AmbientBindingContext");
+            var ambientBindingContext = propertyInfo.GetValue(bindingSource);
+            propertyInfo = ambientBindingContext.GetType().GetProperty("BindingData");
```
Comment (davidebbo) [inline src/WebJobs.Script/Description/ScriptFunctionInvokerBase.cs]: As long as you know you're going to hell for that one ;)

File: src/src.ruleset [hunk 2]
```diff
@@ -16,6 +16,7 @@
     <Rule Id="CA904" Action="None" />
     <Rule Id="CA905" Action="None" />
     <Rule Id="CA908" Action="None" />
+    <Rule Id="CA1062" Action="None" />
```
Comment (davidebbo) [inline src/src.ruleset]: Is that the one I was complaining about?

File: src/WebJobs.Script/Description/ScriptFunctionInvokerBase.cs (hunk 1 above)
Comment (mathewc) [inline src/WebJobs.Script/Description/ScriptFunctionInvokerBase.cs]: I created https://github.com/Azure/azure-webjobs-sdk-script/issues/76 to track this. We'll remove this.

File: src/src.ruleset (hunk 2 above)
Comment (mathewc) [inline src/src.ruleset]: Yep. I aim to please :)

File: src/WebJobs.Script/Description/ScriptFunctionInvokerBase.cs (hunk 1 above)
Comment (fabiocav) [inline src/WebJobs.Script/Description/ScriptFunctionInvokerBase.cs]: :smiling_imp:

File: src/src.ruleset (hunk 2 above)
Comment (fabiocav) [inline src/src.ruleset]: :angel:

**Comment by fabiocav** [comment]
Just happy I got to use :smiling_imp: and :angel: on the same PR

File: src/WebJobs.Script/Description/ScriptFunctionInvokerBase.cs (hunk 1 above)
Comment (davidebbo) [inline src/WebJobs.Script/Description/ScriptFunctionInvokerBase.cs]: And sure enough, that's where it blows up! ;)

**Pull request merged**